### PR TITLE
Multiple failed test fixes

### PR DESCRIPTION
**Pull request by markmcd**
* FareAdapter was failing when it hit unknown keys
* A directions API test with transit details was failing due to unexpected test data

### CODE REVIEW

**Comment by markmcd** [comment]
PTAL @domesticmouse and/or @samthor

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 1]
```diff
@@ -321,13 +322,23 @@ public void testTransitDetails() throws Exception {
...
-    assertNotNull(routes[0].legs[0].steps[0].transitDetails.departureTime);
-    assertNotNull(routes[0].legs[0].steps[0].transitDetails.line);
-    assertNotNull(routes[0].legs[0].steps[0].transitDetails.line.agencies);
-    assertNotNull(routes[0].legs[0].steps[0].transitDetails.line.vehicle);
+    DirectionsLeg testLeg = routes[0].legs[0];
+
+    // Skip the initial walking steps
+    int i = 0;
+    for (; i < testLeg.steps.length - 1
+        && testLeg.steps[i].travelMode != TravelMode.TRANSIT; i++)
```
Comment (samthor) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: Just write this out: look for the `transitIndex`, or something.

```java
int transitIndex = -1;
for (int i = 0; i < testLeg.steps.length; ++i) {
  if (...) {
    transitIndex = i;
    break;
  }
}
assertNotEqual(transitIndex, -1, ...);
```

Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: Done

**Comment by samthor** [comment]
[LGTM](http://lgtm.in/i/yJuXUcBWp)

[:+1:](http://lgtm.in/u/yJuXUcBWp)

**Comment by domesticmouse** [comment]
Post-hummus LGTM

**Pull request merged**